app/controllers/papers_controller.rb

```diff
@@ -221,7 +221,18 @@ def show
     # Don't show the paper to anyone other than the submitting author or an
     # admin.
     if @paper.invisible?
-      head 404 and return unless can_see_hidden_paper?(@paper)
+      # Redirect to login if not logged in
+      if !current_user
+        session[:redirect_to] = request.fullpath
+        redirect_to 'sessions#create', allow_other_host: true and return
```

Comment:
I'm not sure if `sessions#create` is how to bounced someone through the auth flow?

Reply:
No, this won't work, with omniauth we can not login via a redirect, we need a POST to `/auth/orcid` so I think our best option is to show a text like "Please login first" with a button to login. If it is OK with you, I can do it tomorrow.